storage/src/main/java/org/apache/kafka/storage/internals/log/LogConfig.java

```diff
@@ -436,7 +375,7 @@ private Optional<Compression> getCompression() {
     }
 
     public RecordVersion recordVersion() {
```

Comment:
Shall we remove this method? It doesn't make sense now that we don't have a log config for record version.